File: Codeforces/71A_WayTooLongWords.py
N = int(input())
results = []

while N != 0:

    string = input()

    if len(string) <= 10:
        results.append(string)
    else:
        count = 0
        for i in range(1, len(string) - 1):
            count += 1
        results.append('%s%i%s' %(string[0], count, string[-1]))

    N -= 1

for r in results:
    print(r)

File: Codeforces/71A_Way-Too-Long-Words.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector <string> results;
    int n;
    cin >> n;

    while (n != 0){
        string word;
        cin >> word;

        if (word.length() > 10) {
            int count = word.length() - 2;
            string new_word = word[0] + to_string(count) + word.back();
            results.push_back(new_word);
        } else {
            results.push_back(word);
        }

        n--;
    }

    for (int i = 0; i < results.size(); i++)
        cout << results[i] << endl;

    return 0;
}
